### ISSUE TITLE
App Store Error!!

### ISSUE THREAD

**Issue by ashirvad358** [jdg/MBProgressHUD#317]
Hello,
 Got the below mail from App Store team. I think, its because of the variable named "labelText". Can you please update the var name in your next build?

Dear developer,

We have discovered one or more issues with your recent delivery for "XXXX". To process your delivery, the following issues must be corrected:

Non-public API usage:

The app references non-public selectors in XXXX: setLabelText:
If method names in your source code match the private Apple APIs listed above, altering your method names will help prevent this app from being flagged in future submissions. In addition, note that one or more of the above APIs may be located in a static library that was included with your app. If so, they must be removed.

**Comment by Vinzius** [comment]
It wouldn't be a lot of work to change the property name, no?

Do you know a pull-request?

**Comment by matej** [comment]
Why would that only happen on some submissions? Everybody should have problems, if it is in fact `MBProgressHUD`. Have you tried renaming the method in your project and re-submitting?

**Comment by Vinzius** [comment]
I didn't have the problem myself. However if it's a "known" possibility, then it wouldn't hurt to change the name, no?

I submitted an app some days ago. I choose to download the project and change the property name to avoid a potential rejection.

**Comment by matej** [comment]
I don't want to introduce breaking API changes for no reason. If I recall correctly, there was a similar report some time ago (different method though) and it turned out to be an issue with a some other code.  

The method will be deprecated in the upcoming release, but the definition will remain there fro a few more versions so people can adopt to the new API. Unless this is proven to be a valid issue. 

https://github.com/matej/MBProgressHUD/blob/v1/MBProgressHUD.h#L404

**Comment by csrking** [comment]
For the record, I submitted a new app this past December that included the MBProgressHUD component and I did not receive and warning or rejection from the Apple reviewers.

**Comment by matej** [comment]
Closing this due to no other reports showing up.

**Comment by iori57** [comment]
My app got rejected today due to the same reason - The app references non-public selectors 'hide' and 'setLabelText'

I searched the whole app for 'setLabelText' and I couldn't find it, not even the deprecated function that was mentioned above. Weird. I am guessing that it might be autogenerating the stub due to MBProgressHud having @synthesize labelText;

So I did two things in MBProgressHud library:
1) Renamed 'labelText' to 'hudLabelText' 
2) Renamed 'hide' to 'hideHud'

With just this two changes, I submit my app to App Store again and this time it went through without any problems :) So I am pretty sure that Apple are checking and rejecting some apps mistakenly because of these two things.
